pkg/deviceshifu/deviceshifubase/deviceshifubase_config.go

```diff
@@ -212,3 +214,111 @@ func newEdgeDeviceRestClient(config *rest.Config) (*rest.RESTClient, error) {
 
 	return exampleRestClient, nil
 }
+
+// init DeviceShifuConfig With default
+func (dsConfig *DeviceShifuConfig) init() error {
+	if err := dsConfig.DriverProperties.init(); err != nil {
+		klog.Errorf("Error initializing DriverProperties, error %s", err.Error())
+		return err
+	}
+
+	if dsConfig.Telemetries == nil {
+		dsConfig.Telemetries = &DeviceShifuTelemetries{}
+	}
+	if err := dsConfig.Telemetries.init(); err != nil {
+		klog.Errorf("Error initializing Telemetries, error %s", err.Error())
+		return err
+	}
+
+	if err := dsConfig.Instructions.init(); err != nil {
+		klog.Errorf("Error initializing Instructions, error %s", err.Error())
+		return err
+	}
+
+	return nil
+}
+
+func (driverProperties *DeviceShifuDriverProperties) init() error {
```

Comment:
as far as I remember init() is a special function in golang.

for configuration initialization, suggest to use load()

Reply:
https://www.developer.com/languages/inti-function-golang/ FYI